### PR TITLE
154 all jobs through subscriptions

### LINKED ISSUE #154: As a dev, I want all jobs to go through subscriptions
Instead of being hardcoded. 

* Subscriptions are configured to start both on findings and CRON
* Built-in subscriptions are editable by the user
* Built-in subscriptions are flagged as such in the ui
* The user can revert to the original parameters

### PR DESCRIPTION
**Pull request by lm-sec**
Allows for all the current workflow logic from Stalker to be implemented as subscriptions. It achieves that partly by having a new `triggerInterval` for the event subscriptions, as now described in the documentation. This interval is the cooldown for an event subscription and a ressource. They are tracked in database. 

The PR also bring the concept of built-in subscriptions. These are yaml file loaded on start if there are no subscriptions present and the database needs to initialize. The built-in subscriptions are shown in the UI. Having these built-in subscriptions adds the `builtIn` field in the model, as well as the `file` field. You can edit these subscriptions and revert them to their default value.

image

Closes #154

### CODE REVIEW

**Review by Aboisier — APPROVED** [review]
LGTM! Very nice to see we are removing special cases! 🚀

File: docs/subscriptions.md [hunk 1]
```diff
@@ -77,21 +79,25 @@ When Stalker finds some information, either through the output of a Job, or thro
 
 An event subscription allows for the customization of Stalker's automation workflow by starting any Job on any Finding. These jobs can be started on specified conditions. Also, the output of the finding can be used as a job input, as well as a condition parameter.
 
+Event subscriptions also have a trigger interval in seconds. This value ensures that a subscription is not triggered too often for the same ressource. It prevents infinite loops and rescanning the same host right away, for instance.
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply docs/subscriptions.md:82]: Updated everything to cooldown. I was on the fence about it when I picked the name, and it was still annoying me, I am glad you mentionned it. Cleaner in yaml too, less camelCase.

File: docs/subscriptions.md [hunk 2]
```diff
@@ -108,11 +114,14 @@ Here is one of the shortest event subscription possible. It reacts to a `Hostnam
...
-> Something similar in behavior is already built in Stalker.
+The `triggerInterval` here represents 23 hours. Therefore, the same hostname can only be resolved as often as every 23 hours through this subscription.
+
+> This subscription is already built in Stalker
 
 ```yaml
-name: My simple subscription
+name: Domain name resolution
 finding: HostnameFinding
+triggerInterval: 82800
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply docs/subscriptions.md:124]: Agreed, now tracked in #194

File: flow_manager/src/modules/database/subscriptions/cron-subscriptions/cron-subscriptions.service.ts [hunk 3]
```diff
@@ -35,19 +40,20 @@ export class CronSubscriptionsService {
...
       jobName: dto.jobName,
       jobParameters: dto.jobParameters,
+      builtIn: false,
     };
     return await this.subscriptionModel.create(sub);
   }
 
   public async getAll() {
-    return await this.subscriptionModel.find({});
+    return await this.subscriptionModel.find({}, '-file');
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply flow_manager/src/modules/database/subscriptions/cron-subscriptions/cron-subscriptions.service.ts:49]: The idea is that it is a bad practice, in general, to show a real backend file name to a user. It tends to help to exploit vulnerabilities such as "arbitrary file reads". It does not really impact Stalker though as it is open source, so you can just go and read the file names if you want. 

However, this information is not useful to the UI ever, and it is almost never useful to the backend, so it makes sense to me to filter it out.

File: stalker_ui/src/assets/red-kite-logo-only-color-circle.png
[the remark this answers is not in the document]
Comment (lm-sec) [reply stalker_ui/src/assets/red-kite-logo-only-color-circle.png:1]: I put it at 50x50, it is much smaller (~900 bytes). I was able to make it even smaller (~ 290 bytes), but we lost the round of the logo and it looked awful.

File: stalker_ui/src/app/api/jobs/subscriptions/event-subscriptions.service.ts [hunk 4]
```diff
@@ -80,4 +85,8 @@ export class EventSubscriptionsService {
     }
     return data;
   }
+
+  public async revert(id: string) {
+    return await firstValueFrom(this.http.put(`${environment.fmUrl}/event-subscriptions/${id}/revert`, {}));
+  }
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply stalker_ui/src/app/api/jobs/subscriptions/event-subscriptions.service.ts:91]: done

File: stalker_ui/src/app/api/jobs/subscriptions/event-subscriptions.service.ts [hunk 5]
```diff
@@ -44,10 +46,12 @@ export class EventSubscriptionsService {
       _id: newSub._id,
       name: newSub.name,
       finding: newSub.finding,
+      triggerInterval: newSub.triggerInterval,
       companyId: newSub.companyId ? newSub.companyId : allCompaniesSubscriptions,
       job: {
         name: newSub.jobName,
       },
+      builtIn: false,
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply stalker_ui/src/app/api/jobs/subscriptions/event-subscriptions.service.ts:54]: done

**Pull request closed without merging**